compiler/src/lnntoamm/Amm.ts

```diff
@@ -17,6 +17,22 @@ export default class Output {
     this.handlers = [];
   }
 
+  toString(): string {
+    let res = '';
+    for (let [ty, constants] of this.constants.entries()) {
+      for (let constName of Object.keys(constants)) {
+        res = res.concat('const ', constName, ': ', ty.ammName, ' = ', constants[constName]);
+      }
+    }
+    for (let eventName of Object.keys(this.events)) {
+      res = res.concat('event ', eventName, ': ', this.events[eventName].ammName);
+    }
+    for (let handler of this.handlers) {
+      res = res.concat(handler);
+    }
+    return res;
+  }
```

Comment:
Believe it or not, but making this an array, `push`ing these pieces into it, and then `join`ing them at the end will be faster.

Reply:
short story for anyone reading this comment:
I had seen people say that `concat` is faster than `join`, which is why I used `concat`. After seeing this comment, @dfellis and I both found separate benchmarks that tested this feature. His benchmark found that my original hypothesis is true, while my benchmark found that it's false. I think we'll just have to benchmark this ourselves...